Source/FakeItEasy/Core/FakeScope.cs

```diff
@@ -13,19 +16,48 @@ namespace FakeItEasy.Core
     internal abstract class FakeScope
         : IFakeScope
     {
+        private const string CurrentScopeLogicalCallContextKey = "FakeItEasy.FakeScope.Current";
+
+        private static readonly RootScope root;
+        private static readonly ConcurrentDictionary<Guid, FakeScope> scopes;
+
+        [SuppressMessage(
+            "Microsoft.Performance", 
+            "CA1810:InitializeReferenceTypeStaticFieldsInline", 
+            Justification = "RootScope constructor requires scopes to be initialized first.")]
         static FakeScope()
         {
-            Current = new RootScope();
+            scopes = new ConcurrentDictionary<Guid, FakeScope>();
+            root = new RootScope();
         }
 
         private FakeScope()
         {
+            this.Id = Guid.NewGuid();
+            FakeScope.scopes.TryAdd(this.Id, this);

```

Comment:
My mistake. Scopes, I think. While I'm complaining about trivialities, if you end up revising these commits, consider addressing the style warnings:
http://teamcity.codebetter.com/viewLog.html?buildId=207272&buildTypeId=bt929&tab=buildLog#_state=123&focus=124
Thanks.